src/rebuild/README.md

```diff
@@ -1,275 +1,150 @@
-
-<a id="10.6.3"></a>
-### No Multi-Rebalance
-
-To simplify the design, multi-rebalance is not supported in this design. The user has to serialize pool-map extending operations.
+#Rebuild Detection
+
+When a target failed, it should be detected promptly and
+notify the pool (Raft) leader. then the leader will exclude
+the target from the pool and trigger the rebuild process
+immediately.
+
+##Current status and longterm goal
+
+Currently, since the raft leader can not exclude the target
+automatically, the sysadmin has to manually exclude the target
+from the pool, which then trigger the rebuild.
+
+In future, the leader should be able to detect the target failure
+promptly and then trigger the rebuild automatically by itself,
+without the help of sysadmin.
+
+##multiple pool and targets rebuild
+
+If there are multiple pools being impacted by the failing
+target, these pools can be rebuilt concurrently.
+
+If there are multiple targets in the same pool failes before the
+rebuild really happenes, then these targets can be rebuilt at the
+same time, otherwise the new failing target rebuilt has to be deferred
+until the current pool rebuilt is finished.
+
+#Rebuild process
+
+The rebuild is divided into 2 phases, scan and pull.
+
+##Scan
+
+Initially, the leader will propate the failure notification
+to all other surviving targets by a collective RPC. Any target
+that receives this RPC will start to scan its own object table
+to determine the object losts data redundancy on the faulty
+target, if it does, then send their IDs and related metadata
+to the Rebuild Targets. As for how to choose the rebuild target
+for faulty target,it will be described in placement/README.md
+
+##pull
+
+Once the rebuild targets get the object list from the scanning
+target, it will pull the data of these objects from other
+replicas and then write data locally. Each target will report
+its rebuild status, rebuilding objects, records, is_finished?
+etc, to the pool leader. Once the leader learned all of targets
+finished its scanning and rebuilding phase, it will notify all targets
+the rebuild has finished, and they can release all of the resources
+holded during rebuild process.
+
+##I/O during rebuild
+
+During rebuild, I/O is almost the same as usual, except two cases
+
+1. Fetch will try to skip the rebuilding target. 
+
+2. Once the new spare target is choosen, client will start updating
+the data both on the new target, in the mean time, rebuild might also
+update the same data from other replicas, then the data might be
+updated twice during the rebuild.
+
+##rebuild resource throttle
+
+During rebuild process, the user can set the throttle to guarantee
+the rebuild will not use more resource than the user setting. The
+user can only set the CPU cycle for now. For example, if the user set
+the throttle to 50, then the rebuild will at most use 50% of CPU
+cycle to do rebuild job. The default rebuild throttle for CPU cycle
+is 30.
+
```

Comment:
maybe can say a little bit about the space limitation for rebuild (DAOS_PROP_PO_SPACE_RB pool property), although not implemented now.

Reply:
Indeed, but this is not implemented yet, so I would rather update it later once we have it to avoid trouble.